README.md

```diff
@@ -115,6 +115,39 @@ Server
 
 
+Multithreading
+========================
+Our multithreading functionality is done through ActiveJob, which employs DelayedJob on the backend.
+ActiveJob gets better after rails 5.2, so keep in mind that there is some functionality that is not perfect, until we can upgrade rails.
+
+Backstory:  Rails runs on a single thread.  Even when you schedule asynchronous operations, s.a. sending a request with HTTParty, that request is executed on the same thread as the rest of the application.  Rails simulates multithreading with creative use of scheduling.
+
+To put something on its own thread, you have to first create the second thread by starting the second worker on the same EC2 server:
+```
+RAILS_ENV=qa bin/delayed_job start
+```
+
+There are several options for implementing the multithreading backend in the code.  Sidekiq and DelayedJob are two of the more popular ones.  We chose DelayedJob, for its relative simplicity.  Each backend implementation can be used on its own, or through ActiveJob hooks (https://edgeguides.rubyonrails.org/active_job_basics.html).
+
+It's a good idea to use ActiveJob.  Even though using DelayedJob directly can give more powerful functionality, using the ActiveJob intermediary will allow future maintainers to switch implementations without rewriting app code.
+
+How do you call a method asynchronously?  Call it like we do here:
+```
+FindAvailableUserBarcodeJob.perform_later(user: self)
+```
+Perform, and perform_later calls are scheduled to go at some schedule you've set up.  
+At this point, if you have not started your second worker (remember delayed_job start?), your call will go on your regular app stack.  Multithreading will not happen.  If you have a second worker going, your asynchronous call will be put on that second worker's thread.  If you have multiple asynchronous calls going at the same time, they will be put on the second worker thread, but be single-threaded (schedule one after another) within that thread's stack.
+
+DelayedJob Logging:
+There are two places you'll see information on the scheduled code runs.  One is the ```delayed_jobs``` database table.  Here, you can see the jobs that are scheduled to be run.  Usually, after a job is completed, its row will be removed from the table.  So be aware that you won't see historical jobs in that table.  DelayedJob configuration is in:
+```
+config > initializers > delayed_job_config.rb
+```
+The second place is the log.  Our is here: ``` tail -500f log/delayed_job.log ``` .
+
+Our asynchronous code is currently used for user barcode creation on account create.  During that process, we need to talk to Sierra one or more times, and those calls can take up time and resources.  For more info on barcodes, see https://confluence.nypl.org/display/DIGTL/User+Barcodes .
```

Comment:
I'd put this further up the README, so people understand quickly the "why" before reading about the "how"

Reply:
ok